### PR TITLE
Add LUA AP_TemperatureSensor bindings

### PR DESCRIPTION
**Pull request by peterbarker**
... and an example

image

Testing https://github.com/ArduPilot/ardupilot/pull/26886

### CODE REVIEW

**Comment by peterbarker** [comment]
@rahulOCR Something odd going on here.  `get_temperature` should be returning degrees-celsius, so the conversion I'm putting in here is incorrect.

Any ideas?

**Comment by rahulOCR** [comment]
`get_temperature` will return in kelvin. can you let me know what odd is going on ?

File: libraries/AP_Scripting/examples/temperature_sensor.lua [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+   simple example of reading a temperature sensor
+--]]
+
+local MAV_SEVERITY = {EMERGENCY=0, ALERT=1, CRITICAL=2, ERROR=3, WARNING=4, NOTICE=5, INFO=6, DEBUG=7}
+
+--[[
+   main update function, called at 1Hz
+--]]
+function update()
+   local temperature_K = temperature_sensor:get_temperature(0)
```
Comment (rahulOCR) [inline libraries/AP_Scripting/examples/temperature_sensor.lua:11]: this returns in degrees calvin

**Comment by peterbarker** [comment]
Rebased on top of @rahulOCR 's fix (thanks!).

File: libraries/AP_Scripting/generator/description/bindings.desc [hunk 2]
```diff
@@ -973,3 +973,9 @@ singleton AP_VisualOdom depends HAL_VISUALODOM_ENABLED
 singleton AP_VisualOdom rename visual_odom
 singleton AP_VisualOdom method healthy boolean
 singleton AP_VisualOdom method quality int8_t
+
+-- ----AP_TemperatureSensor Library----
+include AP_TemperatureSensor/AP_TemperatureSensor.h
+singleton AP_TemperatureSensor depends AP_TEMPERATURE_SENSOR_ENABLED
+singleton AP_TemperatureSensor rename temperature_sensor
+singleton AP_TemperatureSensor method get_temperature boolean float'Null uint8_t 0 ud->num_instances()
```
Comment (IamPete1) [inline libraries/AP_Scripting/generator/description/bindings.desc:981]: We might need to add some checks in `AP_TemperatureSensor` but we have been trying to not limit instances by `ud->num_instances()` like this. The problem is that a script has no way to know `ud->num_instances()` at runtime. A too large value will cause the script to error out (unless you pcall).

Comment (IamPete1) [reply libraries/AP_Scripting/generator/description/bindings.desc:981]: Looks like its safe to call with any value. 
```suggestion
singleton AP_TemperatureSensor method get_temperature boolean float'Null uint8_t'skip_check
```

Comment (peterbarker) [reply libraries/AP_Scripting/generator/description/bindings.desc]: Done and re-tested, thanks!

**Review by IamPete1 — APPROVED** [review]
LGTM

**Comment by peterbarker** [comment]
Thanks @IamPete1 , merging.

**Pull request merged**